src/torch_tomogram/tomogram.py

```diff
@@ -77,63 +166,116 @@ def extract_particle_tilt_series(
             image=self.images,
             positions=projected_yx,
             sidelength=sidelength,
+            return_rfft=return_rfft,
+            decenter=return_rfft,  
         )
         return images
 
     def reconstruct_subvolume(
         self, point_zyx: torch.Tensor, sidelength: int
     ) -> torch.Tensor:
-        """Reconstruct a subvolume at a 3D location in the sample."""
-        point_zyx = torch.as_tensor(point_zyx, device=self.device).float()
-        point_zyx = point_zyx.reshape((-1, 3))
-        rotation_matrices = self.projection_matrices[:, :3, :3]
-        rotation_matrices = torch.linalg.pinv(rotation_matrices)
-        sidelength_padded = int(self._pad_factor * sidelength)
-        particle_tilt_series = self.extract_particle_tilt_series(
-            point_zyx, sidelength=sidelength_padded
-        )
-        volume = backproject_2d_to_3d(
-            images=particle_tilt_series[0],
-            rotation_matrices=rotation_matrices,
-            zyx_matrices=True,
-            pad_factor=1.0,  # we already incorporate padding in subtilts
-            fftfreq_max=0.5,
-        )
-        p = (sidelength_padded - sidelength) // 2
-        volume = F.pad(volume, [-p] * 6)  # remove padding
-        return volume
+        """Reconstruct a 3D patch at a location in the sample."""
+        # Use batched method and extract first result
+        patches = self.reconstruct_patches_batched(point_zyx, sidelength)
+        return patches[0]
 
     def reconstruct_tomogram(
         self, volume_shape: tuple[int, int, int], sidelength: int
     ) -> torch.Tensor:
-        """Reconstruct the full tomogram by tiling the positions in 3D."""
+        """Reconstruct the full tomogram by tiling reconstructed patches in 3D. """
         d, h, w = volume_shape
         r = sidelength // 2
 
-        # setup grid points
-        z = torch.arange(start=r, end=d + r, step=sidelength) - d // 2
-        y = torch.arange(start=r, end=h + r, step=sidelength) - h // 2
-        x = torch.arange(start=r, end=w + r, step=sidelength) - w // 2
+        # Setup grid points where patches will be reconstructed
+        z = torch.arange(start=r, end=d + r, step=sidelength, device=self.device) - d // 2
+        y = torch.arange(start=r, end=h + r, step=sidelength, device=self.device) - h // 2
+        x = torch.arange(start=r, end=w + r, step=sidelength, device=self.device) - w // 2
 
-        # allocate whole volume
-        tomogram = torch.zeros(
-            size=volume_shape, dtype=torch.float32, device=self.device
+        # Create grid of all positions: (n_z, n_y, n_x, 3)
+        grid_zyx = torch.stack(torch.meshgrid(z, y, x, indexing='ij'), dim=-1)
+        original_shape = grid_zyx.shape[:-1]  # (n_z, n_y, n_x)
+        grid_zyx = grid_zyx.reshape(-1, 3)  # (N, 3) where N = n_z * n_y * n_x
```

Comment:
could use [einops.pack](https://einops.rocks/4-pack-and-unpack/#auto-batching) here - quite nice for making tensor operations rank polymorphic.... would actually be nicer to implement within reconstruct_patches_batched  so the API would be
```
- points_zyx (..., 3)
- output (..., d, h, w)
```
```suggestion
        # Create grid of all center points of reconstructed subvolumes
        centers_zyx = torch.stack(torch.meshgrid(z, y, x, indexing='ij'), dim=-1)  # (gd, gh, gw, 3)
        centers_zyx, ps = einops.pack([centers_zyx], "* zyx")
        ...
        [reconstructions] = unpack(reconstructions, ps, "* d h w")
```